### ISSUE TITLE
output file of video with 0 events is still written

### ISSUE THREAD

**Issue by ZacTheHac** [Breakthrough/DVR-Scan#50]
Basically I have a script that iterates through a few thousand files and outputs motion-only files separately.
The problem is that some of the clips have no events of interest to me, but they still write an 11kb output file.

I feel that if there are zero events, then the output file should be erased (as I know the file is created as soon as the command is run, and then written to at the end)

My current workaround is sorting by file size and deleting everything under 100kb, but it's just an odd quirk.

Also I just noticed the output says "Scan-only mode specified, list of motion events:" even if I have an output file specified, so maybe remove that first half if there's an output specified.

**Comment by Breakthrough** [comment]
Hey @ZacTheHac;

Thanks for the report - this is an edge case that wasn't considered, so will mark it as a bug.  This should be a relatively easy fix, although I'm glad you at least have a workaround for now.

As for the last thing you mentioned about the output, would you be able to share how you're invoking `dvr-scan`?

Thank you!

**Comment by ZacTheHac** [comment]
As for how I'm invoking DVR-Scan, I'm invoking it via PowerShell, specifically by the command:
`get-item *F.MP4 | foreach {Invoke-Expression("dvr-scan -roi 300 500 200 520 -i "+$_.Name+" -o Motion\"+$_.Name.Substring(0,20)+"_Motion.avi")  } `
Which basically runs the command:
`dvr-scan -roi 300 500 200 520 -i *filename* -o Motion\*filename*.avi`
for every file ending in F.MP4 in the folder. (I know I can run them all as one command, but I have a specific use case why I want them separate)

Running the command directly results in the same output.

Here's a screenshot showing it not being said on a video with no events, and a video with events having the "Scan-only mode specified"

![image](https://user-images.githubusercontent.com/7144613/117536230-bd4c5a00-afae-11eb-8db5-284818ff5c8c.png)

**Comment by Breakthrough** [comment]
This is now fixed in master, and will be included in the next release of DVR-Scan.

**Comment by ocram** [comment]
Is this really supposed to be fixed, as indicated here, and included in v1.3, as the commit from May 13 shows?

For me, the output files are still always created immediately, with a size of 0 bytes, and remain there (empty) if no motion is found.

A basic command that produces this behavior is:

```
dvr-scan --input "my-input-video.mp4" --output "my-output-video.avi"
```

DVR-Scan: v1.3 (latest release)
Python: v3.5.1
cv2: v4.2.0
numpy: v1.18.5
OS: Ubuntu 20.04
pip source: `https://bootstrap.pypa.io/pip/3.5/get-pip.py`

**Comment by Breakthrough** [comment]
@ocram I think I see the issue now:
https://github.com/Breakthrough/DVR-Scan/blob/master/dvr_scan/cli.py#L294

I'll need to change that logic up when you specify -o/--output, since that will create an empty file right away.

Thanks for the update, will re-open the issue until that's resolved.
